### ISSUE TITLE
[bug] CMake INSTALL_RPATH setting runtime paths for static libraries linked with zlib, yajl, sqlite, libxml, and hiredis on macOS binaries

### ISSUE THREAD

**Issue by hoyhoy** [conan-io/conan#15875]
### Describe the bug

Not sure if this is a conan-center issue with these particular recipes or an issue with conan itself...

We have these configured with `shared=False`, but on macOS, I'm seeing...

otool -l bin/application...
```
rpath .conan2/p/b/opens4ccf01ac05b40/p/lib (offset 12)
rpath .conan2/p/b/hired42eacbbf680eb/p/lib (offset 12)
rpath .conan2/p/b/yajlbb34de24ea2f0/p/lib (offset 12)
rpath .conan2/p/b/sqlit64322ca61290e/p/lib (offset 12)
rpath .conan2/p/b/libxm557f06d07edf8/p/lib (offset 12)
rpath .conan2/p/b/zlib4e956a7db8658/p/lib (offset 12)
```

Only happens if cmake `BUILD_WITH_INSTALL_RPATH` is enabled.

### How to reproduce it

Build a binary with one of these packages as a static library on MacOS with `BUILD_WITH_INSTALL_RPATH=ON`

**Comment by memsharded** [comment]
Hi @hoyhoy 

Thanks for your report. 

Just to make sure, can you please confirm that you can reproduce this with a simple minimal repro case.
In this case, if we are talking about ConanCenter recipes:

```bash
$ git clone  https://github.com/conan-io/conan-center-index
$ cd conan-center-index/recipes/zlib/all
$ conan create .
$ otool -l test_package/build/<path-to-exe>
```

Does this produce the above information about rpaths?

**Comment by jcar87** [comment]
hi @hoyhoy  thanks for reporting this.

CMake should add the directories to the RPATH in the _build_ tree, but not in the install tree (or conversely, not when `CMAKE_BUILD_WITH_INSTALL_RPATH` is used) - Conan doesn't do anything specific, we run with modern CMake defaults.

I'm unable to reproduce this issue - is it possible that you are missing the `CMAKE_` prefix when defining that variable?
if I pass `-DCMAKE_BUILD_WITH_INSTALL_RPATH=ON`, the resulting executables in the build tree will have the rpaths stripped - so there are none (unless specified with `CMAKE_INSTALL_RPATH`).

**Comment by hoyhoy** [comment]
We specified one path from our build tree in `CMAKE_INSTALL_RPATH`.

I'm seeing that list of _some_ of the package requirements from our conanfile emitting a ` -Wl,-rpath`.  I wouldn't think that would happen with a static library.  We set -`-o *:shared=False`.  It's odd these would be emitted in the first place if we're building static libraries.  We also reproduced this on XCode's clang and clang 18.1.1.

It could be something wrong with the individual recipes.  I know the OpenSSL recipe is very odd indeed because it injects it custom target module cmake from the recipe itself.  But, the zlib recipe is pretty simple.  Nor sure why that would have the issue.  I'm linking against libcurl as well, but that doesn't appear to show up as an rpath.  That makes me think it's recipe related.  It doesn't happen on Linux.  It's macOS / clang only, and only when we have a `CMAKE_BUILD_WITH_INSTALL_RPATH=ON` with a local `CMAKE_INSTALL_RPATH`

**Comment by jcar87** [comment]
We would need a reproducible example, Im afraid.

>  I wouldn't think that would happen with a static library.

CMake *will* add rpaths even for static libraries in the *build* tree - I've tested this building zlib from source locally, completely independent from Conan. The executable still has the path to zlib in the RPATH, even if it's not needed. In the *install* tree (or with `CMAKE_BUILD_WITH_INSTALL_RPATH`) there are no such entries.

So I think we need an example to determine what is going on - I feel like there's different issues:
1. "RPATHs in executable for static libraries" -> as far as I can see, this is done by CMake, Conan does not do this explicitly. This should not hurt as long as is limited to the build tree. I've tested this with ZLIB outside of CMake.
2. RPATHS pointing to directories inside the Conan cache happen in the install tree - this I'm unable to reproduce, I can see CMake removing the paths correctly. When `CMAKE_BUILD_WITH_INSTALL_RPATH` is `ON` there shoudn't be rpaths pointing to conan directories in the executables.
3. Recipes emitting `-Wl,-rpath` - they don't, and they shouldn't - except for limited cases with if it's relative rpaths.


if (2) or (3) are an issue, please let us know. (1), as far as I'm concerned, is at most harmless.

**Comment by hoyhoy** [comment]
Yeah, something is emitting the `-Wl,-path` lines.  I haven't checked if it's conan or the `FindZlib.cmake` files.  It's likely recipe or cmake `find_package` related since it only happens for _some_ of the packages, not all.
